filebeat/harvester/reader/encoder.go

```diff
@@ -0,0 +1,35 @@
+package reader
+
+import (
+	"io"
+	"time"
+
+	"github.com/elastic/beats/filebeat/harvester/encoding"
+)
+
+// Encode reader produces lines by reading lines from an io.Reader
+// through a decoder converting the reader it's encoding to utf-8.
+type Encode struct {
```

Comment:
I think Encode is still correct, but we could name it `EncodeUTF8` to make it more clear?